requestfinder.py

```diff
@@ -0,0 +1,226 @@
+                devel = pm.find('devel') if pm else None
+                if devel is None or devel.get('project') == self.devel:
+                    self.requests.append(req)
+            else:
+                self.requests.append(req)
+
+        return self.requests
+
+
+class CommandLineInterface(ToolBase.CommandLineInterface):
+
+    def __init__(self, *args, **kwargs):
+        ToolBase.CommandLineInterface.__init__(self, args, kwargs)
+
+        self.cp = ConfigParser()
+        d = load_first_config('opensuse-release-tools')
+        if d:
+            self.cp.read(os.path.join(d, 'requestfinder.conf'))
+
+    def get_optparser(self):
+        parser = ToolBase.CommandLineInterface.get_optparser(self)
+        parser.add_option('--devel', dest='devel', metavar='PROJECT',
+                          help='only packages with devel project')
+        return parser
+
+    def setup_tool(self):
+        tool = Requestfinder()
+        tool.devel = self.options.devel
+        return tool
+
+    def _load_settings(self, settings, name):
+        section = 'settings {}'.format(name)
+        for option in settings.keys():
+            if self.cp.has_option(section, option):
+                settings[option] = self.cp.get(section, option).replace('\n', ' ')
+
+    @cmdln.option('--exclude-project', metavar='PROJECT', action='append', help='exclude review by specific project')
+    @cmdln.option('--exclude-user', metavar='USER', action='append', help='exclude review by specific user')
+    @cmdln.option('--query', metavar='filterstr', help='filter string')
+    @cmdln.option('--action', metavar='action', help='action (accept/decline)')
+    @cmdln.option('--settings', metavar='settings', help='settings to load from config file')
+    @cmdln.option('-m', '--message', metavar="message", help="message")
+    def do_review(self, subcmd, opts):
+        """${cmd_name}: print commands for reviews
+
+        ${cmd_usage}
+        ${cmd_option_list}
+        """
+
+        settings = {
+            'action': 'accept',
+            'message': 'ok',
+            'query': None,
+            'exclude-project': None,
+            'exclude-user': None,
+        }
+
+        if opts.settings:
+            self._load_settings(settings, opts.settings)
+
+        if opts.action:
+            settings['action'] = opts.action
+            settings['message'] = opts.action
+
+        if opts.message:
+            settings['message'] = opts.message
+
+        if opts.query:
+            settings['query'] = opts.query
+
+        if not settings['query']:
+            raise Exception('please specify query')
+
+        rqs = self.tool.find_requests(settings['query'])
+        for r in rqs:
+            if r.actions[0].type == 'submit':
+                # only works with full history r.get_creator(),
+                print '#', r.reqid, r.actions[0].type, r.actions[0].src_project, r.actions[0].src_package, r.actions[0].tgt_project
```

Comment:
Old-style (keyword instead of function) `print` is deprecated, but perhaps this code will never live to see python 3.